Bug: Type name functions in interpreter not updated after #904

## Description

After #904 fixed `getEZTypeName()` in `pkg/stdlib/builtins.go`, two similar functions in the interpreter package were not updated:

- `pkg/interpreter/builtins.go:26` - `getEZTypeName()`
- `pkg/interpreter/evaluator.go:3188` - `objectTypeToEZ()`

These functions still fall through to internal Go type names for types like Map, Range, FileHandle, Database, and Reference.

## Example

```
error[E5012]: return type mismatch: expected map[string:string], got MAP
```

The `MAP` should be `map[string:string]`.

## Fix

Update the two interpreter functions to match the complete version in `pkg/stdlib/builtins.go`, or consolidate into a single shared function.

**Missing cases in both:**
- `*Map` → should return `"map[" + v.KeyType + ":" + v.ValueType + "]"`
- `*Range` → should return `"Range<int>"`
- `*FileHandle` → should return `"File"`
- `*Database` → should return `"Database"`
- `*Reference` → should return `"Ref<" + innerType + ">"`

**Additionally missing in `objectTypeToEZ`:**
- `*Char` → should return `"char"`
- `*Byte` → should return `"byte"`

## Related

- #900 - Original issue
- #904 - Fix for stdlib version
- #945 - Duplicate of the Map case specifically

--- a/pkg/interpreter/builtins.go
+++ b/pkg/interpreter/builtins.go
@@ -59,6 +59,17 @@ func getEZTypeName(obj Object) string {
 		return "nil"
 	case *Function:
 		return "function"
+	case *Range:
+		return "Range<int>"
+	case *FileHandle:
+		return "File"
+	case *Database:
+		return "Database"
+	case *Reference:
+		if inner, ok := v.Deref(); ok {
+			return "Ref<" + getEZTypeName(inner) + ">"
+		}
+		return "Ref"
 	default:
 		return string(obj.Type())
 	}

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -3195,6 +3195,10 @@ func objectTypeToEZ(obj Object) string {
 		return "string"
 	case *Boolean:
 		return "bool"
+	case *Char:
+		return "char"
+	case *Byte:
+		return "byte"
 	case *Array:
 		// Return typed array format if element type is known
 		if v.ElementType != "" {
@@ -3231,6 +3235,17 @@ func objectTypeToEZ(obj Object) string {
 			return "map[" + v.KeyType + ":" + v.ValueType + "]"
 		}
 		return "map"
+	case *Range:
+		return "Range<int>"
+	case *FileHandle:
+		return "File"
+	case *Database:
+		return "Database"
+	case *Reference:
+		if inner, ok := v.Deref(); ok {
+			return "Ref<" + objectTypeToEZ(inner) + ">"
+		}
+		return "Ref"
 	default:
 		return string(obj.Type())
 	}

--- a/pkg/interpreter/object.go
+++ b/pkg/interpreter/object.go
@@ -43,6 +43,8 @@ type (
 	Reference       = object.Reference
 	Range           = object.Range
 	TypeValue       = object.TypeValue
+	FileHandle      = object.FileHandle
+	Database        = object.Database
 )
 
 // Re-export constants
@@ -69,6 +71,8 @@ const (
 	REFERENCE_OBJ    = object.REFERENCE_OBJ
 	RANGE_OBJ        = object.RANGE_OBJ
 	TYPE_OBJ         = object.TYPE_OBJ
+	FILE_HANDLE_OBJ  = object.FILE_HANDLE_OBJ
+	DATABASE_OBJ     = object.DATABASE_OBJ
 
 	// Visibility constants
 	VisibilityPublic  = object.VisibilityPublic

